client/broadcast.go

```diff
@@ -142,3 +146,38 @@ func (ctx Context) BroadcastTxAsync(txBytes []byte) (*sdk.TxResponse, error) {
 
 	return sdk.NewResponseFormatBroadcastTx(res), err
 }
+
+// TxServiceBroadcast is a helper function to broadcast a Tx with the correct gRPC types
+// from the tx service. Calls `clientCtx.BroadcastTx` under the hood.
+func TxServiceBroadcast(grpcCtx context.Context, clientCtx Context, req *tx.BroadcastTxRequest) (*tx.BroadcastTxResponse, error) {
+	if req == nil || req.TxRaw == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid empty tx")
+	}
+
+	clientCtx = clientCtx.WithBroadcastMode(normalizeBoradcastMode((req.Mode)))
+	txBytes, err := proto.Marshal(req.TxRaw)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := clientCtx.BroadcastTx(txBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tx.BroadcastTxResponse{
+		TxResponse: resp,
+	}, nil
+}
+
+func normalizeBoradcastMode(mode tx.BroadcastMode) string {
```

Comment:
```suggestion
func normalizeBroadcastMode(mode tx.BroadcastMode) string {
```

Reply:
nit: `BroadcastMode` should implement `fmt.Stringer` so we can can just call `req.Mode.String()`

Reply:
I tried that, but gogo's custom stringer doesn't play well with grpc-gateway (unmatchable enum value). I reverted this in 57cc0e8fa1e9fdb22bf8690fc08090022faf5c98